client/app/components/Filters.jsx

```diff
@@ -0,0 +1,122 @@
+import { isArray, map, find, includes, every, some } from 'lodash';
+import moment from 'moment';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import Select from 'antd/lib/select';
+
+const ALL_VALUES = '###Redash::Filters::SelectAll###';
+const NONE_VALUES = '###Redash::Filters::Clear###';
+
+export const FilterType = PropTypes.shape({
+  name: PropTypes.string.isRequired,
+  friendlyName: PropTypes.string.isRequired,
+  multiple: PropTypes.bool,
+  current: PropTypes.oneOfType([
+    PropTypes.any,
+    PropTypes.arrayOf(PropTypes.any),
+  ]).isRequired,
+  values: PropTypes.arrayOf(PropTypes.any).isRequired,
+});
+
+export const FiltersType = PropTypes.arrayOf(FilterType);
+
+function createFilterChangeHandler(filters, onChange) {
+  return (filter, value) => {
+    if (filter.multiple && includes(value, ALL_VALUES)) {
+      value = [...filter.values];
+    }
+    if (filter.multiple && includes(value, NONE_VALUES)) {
+      value = [];
+    }
+    const allFilters = map(filters, f => (f.name === filter.name ? { ...filter, current: value } : f));
+    const changedFilter = find(allFilters, f => f.name === filter.name);
+    onChange(allFilters, changedFilter);
```

Comment:
Doesn't look like anyone is consuming this second argument.

Reply:
Probably I was using it at some point during development, but later it became redundant. Will remove it.